Source/Core/DolphinWX/VideoConfigDiag.cpp

```diff
@@ -748,3 +752,30 @@ void VideoConfigDiag::PopulatePostProcessingShaders()
 	postprocessing_shader.LoadShader(vconfig.sPostProcessingShader);
 	button_config_pp->Enable(postprocessing_shader.HasOptions());
 }
+
+void VideoConfigDiag::OnSSAAClick(wxCommandEvent& event)
+{
+	// Check the checkbox status and not the config option because config hasn't changed yet.
+	vconfig.bSSAA = ssaa_checkbox->IsChecked();
+	RefreshAAList();
+}
+
+void VideoConfigDiag::RefreshAAList()
+{
+	choice_aamode->Clear();
+	const std::string& suffix = vconfig.bSSAA ? "x SSAA" : "x MSAA";
+
+	for (std::string mode : vconfig.backend_info.AAModes)
```

Comment:
As far as I could see here, ```backend_info.AAModes``` is populated with mere numbers by all the backends; except for the "None" entry.
How about changing that to be a list of samples (or whatever that number stands for), only adding the ones actually being supported (such as 2, 4, 8, 16) and implying "None" (unless there is a backend that does not support "No xSAA" or "0x xSAA"). This would greatly simplify ```DX11::D3D::EnumAAModes```, ```DX11::D3D::InitBackendInfo``` and this method here; along with removing the duplicated "None" translation from all the backends (which would be moved here, in front of the loop, with an explicit translation call rather than an indirect one).